tests/packet_tests.cpp

```diff
@@ -359,4 +359,154 @@ BOOST_AUTO_TEST_CASE(PacketValueConstructorRequest)
     BOOST_TEST(values == d, boost::test_tools::per_element());
 }
 
+BOOST_AUTO_TEST_CASE(PacketBufferCopyConstructorRequest)
```

Comment:
No need to say 'Buffer' here. It does not matter how you construct your original packet. You could have used value constructor as well, it's all the same.